crates/network-monitor/src/log_queue.rs

```diff
@@ -0,0 +1,60 @@
+use std::collections::VecDeque;
+use std::mem;
+use std::sync::Mutex;
+
+/// Thread-safe VecDeque which can be drained (consumed until empty) without blocking writers.
+pub(crate) struct LogQueue<T> {
+    entries: Mutex<VecDeque<T>>,
+}
+
+impl<T> LogQueue<T> {
+    pub(crate) fn new() -> Self {
+        LogQueue {
+            entries: Mutex::new(VecDeque::new()),
+        }
+    }
+
+    pub(crate) fn write(&self, data: T) {
+        let mut entries = self.entries.lock().expect("Couldn't get LogQueue lock");
```

Comment:
nitpick: Rust errors should not be capitalized. Also, I suggest to stay concise for this specific failure point, explaining we got a poisoned mutex (highly unlikely, since we should not panic while holding the lock as far as I can see).

```suggestion
        let mut entries = self.entries.lock().expect("poisoned");
```